gcpbuildcache/src/main/kotlin/androidx/build/gradle/gcpbuildcache/GcpStorageService.kt

```diff
@@ -165,8 +173,67 @@ internal class GcpStorageService(
                 .build()
         }
 
+        /**
+         * Attempts to use reflection to clear the cached credentials inside the Google authentication library.
+         */
+        private fun clearCachedDefaultCredentials() {
+            try {
+                val field = GoogleCredentials::class.java.getDeclaredField("defaultCredentialsProvider")
+                field.isAccessible = true
+                val defaultCredentialsProvider = field.get(null)
+                println(defaultCredentialsProvider.toString() + " " + field.type.name)
+                val cachedCreds = field.type.getDeclaredField("cachedCredentials")
+                cachedCreds.isAccessible = true
+                cachedCreds.set(defaultCredentialsProvider, null)
+            } catch (exception: Exception) {
+                // unable to clear the credentials, oh well.
+            }
+        }
+
+        private fun defaultApplicationGcpCredentials(
+            scopes: List<String>,
+            messageOnAuthenticationFailure: String,
+            forceClearCache: Boolean
+        ): GoogleCredentials {
+            if (forceClearCache) clearCachedDefaultCredentials()
+            val credentials = GoogleCredentials.getApplicationDefault().createScoped(scopes)
+
+            try {
+                // If the credentials have expired,
+                // reauth is required by the user to be able to generate or refresh access token;
+                // Refreshing the access token here helps us to provide a useful error message to the user
+                // in case the credentials have expired
+                credentials.refreshIfExpired()
+            } catch (e: Exception) {
+                if (!forceClearCache) {
```

Comment:
Shouldn't this be if (forceClearCache)? I don't think we'll go in the else block ever otherwise.

Reply:
Yeah good catch. Fixed.